editoast/editoast_derive/src/error.rs

```diff
@@ -62,6 +74,62 @@ pub fn expand_editoast_error(input: &DeriveInput) -> Result<TokenStream> {
     })
 }
 
+fn parse_error_definition(
+    base_id: &String,
+    default_status: u16,
+    variant: &ParsedVariant,
+) -> Result<TokenStream> {
+    //Error name
+    let name = variant.ident.unraw().to_string();
+
+    // Compute its id
+    let id = format!("editoast:{}:{}", base_id, name);
+
+    // Retieve error status (or get the default one)
+    let status = match variant.params.status.as_ref() {
+        Some(syn::Expr::Lit(exprlit)) => match &exprlit.lit {
+            Lit::Int(lit) => lit.base10_parse::<u16>().unwrap(),
+            _ => default_status,
+        },
+        _ => default_status,
+    };
+
+    // Retrieve the list of parameters that are given in the error
+    let mut context = std::collections::HashMap::new();
+    let untyped_prop: Vec<String> = variant
+        .fields
+        .iter()
+        .filter_map(|field| match field.ident.as_ref() {
+            Some(name) => {
+                let prop_name = name.to_string();
+                match extract_type(&field.ty) {
+                    Some(prop_type) => {
+                        context.insert(prop_name, prop_type);
+                        None
+                    }
+                    _ => Some(prop_name),
+                }
+            }
+            _ => None,
+        })
+        .collect::<Vec<String>>();
```

Comment:
You already helped the compilator line 99.
```suggestion
        .collect();
```